nonce.go

```diff
@@ -74,9 +80,20 @@ type Nonce struct {
 	value []byte
 }
 
-// NewNonce returns a Nonce initialized with the supplied byte slice
-func NewNonce(v []byte) *Nonce {
-	return &Nonce{v}
+// NewNonce returns a Nonce initialized with the supplied byte slice or an error
+// if the supplied buffer is either too big (more than 64 bytes) or too small
+// (less than 8 bytes)
+func NewNonce(v []byte) (*Nonce, error) {
+	nonceSize := len(v)
+	if nonceSize < MinNonceSize || nonceSize > MaxNonceSize {
+		return nil,
+			fmt.Errorf(
+				"invalid nonce size %d, must be in range [%d, %d]",
+				nonceSize, MinNonceSize, MaxNonceSize,
+			)
+	}
+
+	return &Nonce{v}, nil
```

Comment:
Perhaps on Line 126 the same constants can now be used?

Reply:
nice catch!

Reply:
done in eb4643ed5ac3961a591471a1c5ba8473d842273e